packages/labs/ssr-dom-shim/src/index.ts

```diff
@@ -207,15 +231,44 @@ const CustomElementRegistryShim = class CustomElementRegistry {
       // returns the constructor).
       observedAttributes: ctor.observedAttributes ?? [],
     });
+    this.__pendingWhenDefineds.get(name)?.resolve(ctor);
+    this.__pendingWhenDefineds.delete(name);
   }
 
   get(name: string) {
     const definition = this.__definitions.get(name);
     return definition?.ctor;
   }
-};
+
+  getName(ctor: CustomHTMLElementConstructor) {
+    for (const [name, definition] of this.__definitions.entries()) {
```

Comment:
I noticed that `define()` isn't implemented correctly - it should throw if the same class is registered with a different name. Since fixing that would require a lookup by by ctor as well, and `define()` is more perf sensitive than `getName()` I think we should probably have a `__definitionsByCtor` map and go ahead and make this method fast too.